std/numeric.d

```diff
@@ -171,8 +171,24 @@ struct CustomFloat(uint             precision,  // fraction bits (23 for float)
                    uint             exponentWidth,  // exponent bits (8 for float)  Exponent width
                    CustomFloatFlags flags,
                    uint             bias)
-if (((flags & flags.signed)  + precision + exponentWidth) % 8 == 0 &&
-    precision + exponentWidth > 0)
+    // Restrictions from bitfield
+if ((((flags & flags.signed) + precision + exponentWidth) == 8 ||
+    ((flags & flags.signed) + precision + exponentWidth) == 16 ||
+    ((flags & flags.signed) + precision + exponentWidth) == 32 ||
+    ((flags & flags.signed) + precision + exponentWidth) == 64 ||
+    precision == 64 &&  // due to CustomFloat!80 support hack, see below
+        ((flags & flags.signed) + exponentWidth == 8 ||
+        (flags & flags.signed) + exponentWidth == 16 ||
+        (flags & flags.signed) + exponentWidth == 32 ||
+        (flags & flags.signed) + exponentWidth == 64)) &&
+    // mantissa needs to fit into real mantissa
+    (precision <= real.mant_dig - 1 || precision == 64) &&
+    // exponent needs to fit into real exponent
+    1L << exponentWidth - 1 <= real.max_exp &&
```

Comment:
This is the same for the exponent: The number of bits in the exponent may not exceed the number of bits in the exponent of real for the same reason. Because there is no property telling the number of bits in the exponent (e) of a type, I resorted to max_exp which is 2^^(e-1). To avoid complications with rounding errors of floating point calculation, I don't use log2() but <<. The tests before allready made sure, that exponentWidth<=64 and therefore no overflow will happen.

Reply:
This broke the unittests for the following now erroneous instantiation, for all targets with a `real` exponent of less than 15 bits, i.e., about all non-x86 and non-AArch64 targets out there + MSVC targets for LDC: https://github.com/dlang/phobos/blob/5eddbcbac2e23c4d72a06d68149d979dd72f9199/std/numeric.d#L623

Reply:
> This broke the unittests for the following now erroneous instantiation, for all targets with a `real` exponent of less than 15 bits, i.e., about all non-x86 and non-AArch64 targets out there + MSVC targets for LDC:

I've got some ideas, but I'm too tired to figure this out now. I'll have a closer look tomorrow. Probably this unittest needs to be removed, when real.max_exp < 15.

Reply:
Okay thx for taking care of this. - `CustomFloat!(1, 7, CustomFloatFlags.ieee ^ CustomFloatFlags.signed)` seems to work as well.

This fails too with 64-bit `real`: https://github.com/dlang/phobos/blob/5eddbcbac2e23c4d72a06d68149d979dd72f9199/std/numeric.d#L1510

Reply:
Funnily, I finally came up with that `CustomFloat!(1, 7, ...)` too. See PR #7328.

The second issue is unrelated to this PR. It's caused by PR #7267. I'll take care of this too.